proposals/dao/fip_99.ts

```diff
@@ -0,0 +1,137 @@
+    addresses.rai, // rai as the underlying token
+    addresses.feiDAOTimelock // surplus target = timelock
+  );
+  await raiPriceBoundPSM.deployed();
+
+  logging && console.log(`Rai price bound psm deployed`);
+
+  const pcvDripControllerFactory = await ethers.getContractFactory('PCVDripController');
+  const raiPCVDripController = await pcvDripControllerFactory.deploy(
+    addresses.core,
+    addresses.aaveRaiPCVDeposit, // drip source = aave rai pcv deposit
+    raiPriceBoundPSM.address, // drip target = rai psm
+    1800, // drip interval = 30 minutes
+    ethers.constants.WeiPerEther.mul(2_000_000), // drip size = 2 million
+    0 // incentive amount = 0
+  );
+
+  logging && console.log(`Rai pcv drip controller deployed`);
+
+  logging && console.log('Rai PCV drip controller and rai price-bound psm deployed');
+
+  return {
+    raiPriceBoundPSM,
+    raiPCVDripController
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in setup for fip${fipNumber}`);
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // Ensure that the global rate limited minter has the MINTER role & the noncustodialpriceboundpsm has the PCV_CONTROLLER role
+  await contracts.core.hasRole(ethers.utils.id('MINTER_ROLE'), addresses.raiPriceBoundPSM);
+  await contracts.core.hasRole(ethers.utils.id('PCV_CONTROLLER_ROLE'), addresses.raiPCVDripController);
+
+  // Ensure that the rai aave pcv deposit has >= 10m RAI
+  expect((await contracts.aaveRaiPCVDeposit.balance()).gte(ethers.constants.WeiPerEther.mul(10_000_000)));
+
+  // Ensure that minting is paused on the rai psm
+  expect(await contracts.raiPriceBoundPSM.mintPaused()).to.equal(true);
+
+  // Ensure that redeem is not paused on the rai psm
+  expect(await contracts.raiPriceBoundPSM.redeemPaused()).to.equal(false);
+
+  // Ensure that the fuse pool 9 rai pcv deposit and the aave rai pcv deposit are whitelisted on the pcv guardian
+  expect(await contracts.pcvGuardian.isSafeAddress(addresses.rariPool9RaiPCVDeposit)).to.equal(true);
+  expect(await contracts.pcvGuardian.isSafeAddress(addresses.aaveRaiPCVDeposit)).to.equal(true);
+
+  // Call drip on the rai pcv drip controller and make sure the rai price bound psm got the drip
+  await contracts.raiPCVDripController.drip();
```

Comment:
this belongs in an e2e test, not the validate function

Reply:
Why is the NonCustodialPSM being dripped to?

Reply:
> Why is the NonCustodialPSM being dripped to?

It's not? We're dripping into the rai pcv deposit. (we're not using the non-custodial psm here)